src/realm/query_expression.hpp

```diff
@@ -3679,7 +3720,8 @@ class SubQuery {
 namespace aggregate_operations {
 template <typename T, typename Derived, typename R = T>
 class BaseAggregateOperation {
-    static_assert(std::is_same<T, Int>::value || std::is_same<T, Float>::value || std::is_same<T, Double>::value,
+    static_assert(std::is_same<T, Int>::value || std::is_same<T, Float>::value || std::is_same<T, Double>::value ||
```

Comment:
Have you considered writing a helper like `is_one_of<Query, Types...>`? It seems like you are doing this in a few places, so this should be more readable. Something like

```cpp
template <typename Query, typename... Types>
struct IsOneOf;
template <typename Query>
struct IsOneOf<Query> : std::false_type { // Base case.
};
template <typename Match, typename... Rest>
struct IsOneOf<Match, Match, Rest...> : std::true_type { // Found it.
};
template <typename Query, typename NoMatch, typename... Rest>
struct IsOneOf<Query, NoMatch, Rest...> : IsOneOf<Query, Rest...> { // Keep looking.
};
```

In c++17 it could be the much simpler `inline constexpr bool IsOneOf = (... || std::is_same_v<Query, Types>);`

Reply:
Turns out we have `realm::is_any` in our utilities, but I forgot about it. And you're right, there's probably a bunch of other places where we can convert to use this pattern for readability.